includes/Core/Util/Google_URL_Matcher_Trait.php

```diff
@@ -84,4 +88,20 @@ protected function strip_url_scheme( $url ) {
 	protected function strip_domain_www( $domain ) {
 		return preg_replace( '/^www\./', '', $domain );
 	}
+
+	/**
+	 * Returns a punycode version of a unicode URL.
+	 *
+	 * @since n.e.x.t
+	 *
+	 * @param string $url The URL to decode.
+	 */
+	protected function decode_unicode_url( $url ) {
+		$parts = wp_parse_url( $url );
+		if ( ! $parts ) {
+			return $url;
+		}
+		$decoded = \Requests_IDNAEncoder::encode( $parts['host'] );
+		return $parts['scheme'] . '://' . $decoded . $parts['path'];
```

Comment:
Let's use individual replacements for the different parts here (i.e. replace the original host with the decoded host in the original `$url`). Also, in the service you also encoded the path, but not here - I think we should make these two consistent as well.

Reply:
I'll work on adding test cases, and adding the (url) encoding for the path. The reason for doing the combined replacement is to avoid situations where the path or domain are duplicated (eg. "http://türkish.com/türkish") which could make the path replacement not work correctly.

Reply:
@adamsilverstein See my comment on the CL - I think we should for now stay away from paths, as that increases the chance for bugs. I think none of us is an expert on the matter to say the least, and the support requests (as well as the GH issue) is only about domains.